resources/cf-url-rewriter/cf-url-rewriter.ts

```diff
@@ -66,3 +71,20 @@ function errorResponse() {
     statusDescription: 'Not found',
   };
 }
+
+function getIndexFilePath(urisplit: string[]): string {
+    const version = urisplit[0].substring(urisplit[0].lastIndexOf("/") + 1);
+    const [major, minor, patch] = version.split('.').map(Number);
+
+    if (
+        (major == 1 && minor == 3 && patch >= 15) ||
+        (major == 2 && minor >= 12) ||
+        (major >= 3)
```

Comment:
It is indeed a tricky problem. The latest patch release for `2.11` is `2.11.1` I think it is theoretically possible to have a newer patch release, say `2.11.2`. Then this logic will fail. The same thing will happen to all existing 2.x releases < 2.12

So we have several options here.

Option 1: extend your current comparison logic to cover all 2.x versions. This will be a long list.

Option 2: back fill S3 buckets for all released versions with the new structure. Then you can remove the comparison. This can simplify the code but require tedious manual effort.

Option 3: always run httpsGet against the new url. If returned empty, then fall back to the old url. Then you do not need to backfill S3 buckets and also do not need to maintain a long comparison list. 

I would prefer Option 3 for simplicity.

Reply:
If you want to get the semver try using https://www.npmjs.com/package/semver library.